scripts/set_icmp_state

```diff
@@ -0,0 +1,293 @@
+
+ICMP_IP_PROTOCOL_NUM = "1"
+ICMPV6_NEXT_HEADER_NUM = "58"
+
+BLOCK_ALL_PRIORITY = "9999"
+BLOCK_DEST_PRIORITY = "9995"
+
+
+def notice(msg):
+    """Log a NOTICE message to the console and syslog."""
+    syslog.syslog(syslog.LOG_NOTICE, msg)
+    print(msg)
+
+
+def error(msg):
+    """Log an ERR message to the console and syslog, and exit the program with an error code."""
+    syslog.syslog(syslog.LOG_ERR, msg)
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+
+def verify_table_exists(configdb, table_name):
+    """Verify that a given ACL table exists in Config DB."""
+    target_table = configdb.get_entry(CONFIG_DB_ACL_TABLE_TABLE, table_name)
+
+    if not target_table:
+        error("Table {} not found, exiting...".format(table_name))
+
+    return target_table
+
+
+def verify_valid_ip_address(ip_address, table_ip_version):
+    """Verify that a given IP address is valid.
+
+    This checks that 1) the provided address is valid, and 2) that the IP version of the
+    address matches the IP version of the table this address will be applied to.
+    """
+    try:
+        destination_ip_version = ipaddress.ip_network(ip_address).version
+        table_ip_version = 6 if table_ip_version == IP_VERSION_V6 else 4
+        if destination_ip_version != table_ip_version:
+            error("Cannot use an IPv{} rule with an IPv{} table, exiting...".format(destination_ip_version, table_ip_version))
+    except Exception as e:
+        error("Could not parse {} as a valid IP address; exception={}".format(ip_address, e))
+
+
+def verify_valid_vlan_id(configdb, vlan_id):
+    """Verify that a given VLAN ID is valid.
+
+    This checks that 1) the provided VLAN ID is valid/in-bounds, and 2) that the VLAN
+    exists in Config DB.
+    """
+    if vlan_id is None:
+        return
+
+    if vlan_id <= 0 or vlan_id >= 4096:
+        error("Vlan ID {} is out of bounds [1, 4095]".format(vlan_id))
+
+    target_vlan = configdb.get_entry(CONFIG_DB_VLAN_TABLE, "Vlan{}".format(vlan_id))
+
+    if not target_vlan:
+        error("Vlan{} not found".format(vlan_id))
+
+
+def get_table_ip_version(configdb, table_name):
+    """Get the IP version of the ACL table.
+
+    Returns "V4" for IPv4 and "V6" for IPv6.
+    """
+    table = verify_table_exists(configdb, table_name)
+
+    table_type = table.get("type", None)
+    if not table_type:
+        error("Table {} has no type, invalid configuration, exiting...".format(table_name))
+
+    return IP_VERSION_V6 if table_type in ("L3V6", "MIRRORV6") else IP_VERSION_V4
+
+
+def get_block_rule_key(destination_ip_address):
+    """Get the key that will be used to refer to the ACL rule used to block ICMP traffic to a destination.
+
+    Since the rules are all given the same priority in SONiC, we can't identify a rule based on the priority.
+    So, we use the destination IP being blocked to give each rule a unique name in the system.
+    """
+    return BLOCK_KEY_PREFIX + str(destination_ip_address)
```

Comment:
>destination_ip_address [](start = 34, length = 22)

Since this is user input, and will later used to query db, do you want to validate it?

Reply:
This is validated in `verify_valid_ip_address`. Are there some other checks I should do as well?

Reply:
I see. That's good validation!

---
In reply to: [591861189](https://github.com/Azure/sonic-utilities/pull/1488#discussion_r591861189) [](ancestors = 591861189)